kubernetes/database.libsonnet

```
@@ -5,13 +5,13 @@ local resources = import 'resources.libsonnet';
   DatabaseCredential(name, app, namespace): k._Object('databases.outreach.io/v1', 'DatabaseCredential', name, app=app, namespace=namespace) {
     username:: error 'username is required',
     vault:: null,
-    iamauth:: false,
+    auth:: 'mysql',
     local this = self,
     spec: {
       username: this.username,
       grants: this.grants,
       vault: this.vault,
-      iamauth: this.iamauth,
+      auth: this.iamauth,
```

Should this be `this.auth` since the field was renamed?

```suggestion
      auth: this.auth,
```

Also, what is the `auth::` hidden field for?  Can we just say `spec.auth: 'mysql'`?

---

from what i understand this sets the default value, so even if the manifest does not have `auth` spec, like in previous manifests with vault creds, it will simply set the default value instead of failing.

---

ok thanks for the clarification